src/main/java/games/strategy/engine/lobby/client/ui/TimespanDialog.java

```diff
@@ -0,0 +1,107 @@
+package games.strategy.engine.lobby.client.ui;
+
+import java.awt.Component;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import javax.swing.JComboBox;
+import javax.swing.JOptionPane;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerNumberModel;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.AncestorEvent;
+import javax.swing.event.AncestorListener;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import swinglib.JLabelBuilder;
+import swinglib.JPanelBuilder;
+
+/**
+ * A UI-Utility class that can be used to prompt the user for a ban or mute time.
+ */
+public class TimespanDialog {
+
+  /**
+   * The possible time units and corresponding mappings.
+   */
+  @VisibleForTesting
+  enum TimeUnit {
+    MINUTES("Minutes", i -> Instant.now().plus(i, ChronoUnit.MINUTES)),
+    HOURS("Hours", i -> Instant.now().plus(i, ChronoUnit.HOURS)),
+    DAYS("Days", i -> Instant.now().plus(i, ChronoUnit.DAYS)),
+    WEEKS("Weeks", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.WEEKS).toInstant(ZoneOffset.UTC)),
+    MONTHS("Months", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.MONTHS).toInstant(ZoneOffset.UTC)),
+    YEARS("Years", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.YEARS).toInstant(ZoneOffset.UTC)),
+    FOREVER("Forever", i -> null);
+
+    private final String name;
+    private final Function<Integer, Instant> function;
+
+    private TimeUnit(final String name, final Function<Integer, Instant> function) {
+      this.name = name;
+      this.function = function;
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+
+    @VisibleForTesting
+    Instant getInstant(final Integer integer) {
+      return function.apply(integer);
+    }
+  }
+
+  /**
+   * Prompts the user to enter a timespan.
+   * If the operation is not cancelled, the action Consumer is run.
+   * Not that the Date passed to the consumer can be null if the user chose forever.
+   */
+  public static void prompt(final Component parent, final String title, final String infoMessage,
+      final Consumer<Date> action) {
+    final JSpinner spinner = new JSpinner(new SpinnerNumberModel(0, 0, 99999999, 1));
```

Comment:
@RoiEXLab, forgot to bump this yesterday...  Just curious if setting the default value of the spinner to 1 instead of 0 would be better UX?  The old code used to default the duration to 1.  If the user just hits <kbd>Enter</kbd> with 0 set, no ban/mute will be occur.  Maybe that was your intention?

Reply:
Idea was to have 0 as some sort of unban method, but there is a seperate menu option for this, so not really needed ^^